sources/packages/backend/apps/api/src/route-controllers/student-scholastic-standings/student-scholastic-standings.institutions.controller.ts

```diff
@@ -139,4 +166,123 @@ export class ScholasticStandingInstitutionsController extends BaseController {
+   * When an error happen it will return all the records (with the error) and
+   * also a user friendly description of the errors to be fixed.
+   */
+  @ApiBadRequestResponse({
+    description:
+      "Error while parsing text file or " +
+      "one or more text data fields received are not in the correct format or " +
+      "there are no records to be imported.",
+  })
+  @ApiUnprocessableEntityResponse({
+    description:
+      "Application withdrawal has invalid data or " +
+      "some error happen with one or more application withdrawal and the entire process was aborted.",
+  })
+  @IsInstitutionAdmin()
+  @IsBCPublicInstitution()
+  @Post("application-bulk-withdrawal")
+  @UseInterceptors(
+    FileInterceptor("file", {
+      limits: uploadLimits(
+        MAX_UPLOAD_FILES,
+        APPLICATION_BULK_WITHDRAWAL_MAX_UPLOAD_PARTS,
+        APPLICATION_BULK_WITHDRAWAL_UPLOAD_MAX_FILE_SIZE,
+      ),
+      fileFilter: textFileFilter,
+    }),
+  )
+  async bulkWithdrawal(
+    @UploadedFile() file: Express.Multer.File,
+    @Query("validation-only", new DefaultValuePipe(false), ParseBoolPipe)
+    validationOnly: boolean,
+  ): Promise<PrimaryIdentifierAPIOutDTO[]> {
+    // Read the entire file content.
+    const fileContent = file.buffer.toString();
+    // Convert the file raw content into text models.
+    let textModels: ApplicationWithdrawalImportTextModel[];
+    try {
+      textModels =
+        this.applicationWithdrawalImportTextService.readText(fileContent);
+    } catch (error: unknown) {
+      if (
+        error instanceof CustomNamedError &&
+        error.name === APPLICATION_WITHDRAWAL_INVALID_TEXT_FILE_ERROR
+      ) {
+        // When there is an error in header and footer record, we are transforming it in the validationResult format.
+        // This can contain either header or footer error, one at a time.
+        const validationResult: ApplicationBulkWithdrawalValidationResultAPIOutDTO =
+          {
+            errors: [error.message],
+            infos: [],
+            warnings: [],
+          };
+        throw new BadRequestException(
+          new ApiProcessError(
+            error.message,
+            APPLICATION_WITHDRAWAL_INVALID_TEXT_FILE_ERROR,
+            [validationResult],
+          ),
+        );
+      }
+    }
+    // Validate the text models.
+    const textValidations =
+      this.applicationWithdrawalImportTextService.validateTextModels(
+        textModels,
+      );
+    // Assert successful validation.
+    this.assertTextValidationsAreValid(textValidations);
+    // TODO Add business validation for application bulk withdrawal.
+
+    if (validationOnly) {
+      // If the endpoint is called only to perform the validation and no error was found
+      // return an empty array because no record will be created.
+      return [];
+    }
+
+    // TODO create a block to do database updates for Application withdrawal.
```

Comment:
after line 254, I would say to add `return[]` to justify the return type of the method.